### ISSUE TITLE
ES2015 Block scope not honored is some situations

### ISSUE THREAD

**Issue by subhaze** [Microsoft/TypeScript#7182]
**TypeScript Version:** 

1.7.5 / 1.8.0-beta / nightly (1.9.0-dev.20160217)

**Code**

```ts
{
  let foo = 'bar';
}

console.log(foo)

```

**Expected behavior:**

I would expect `let foo = 'bar';` to be emitted as `var foo_1 = 'bar';`

When running the emitted code I would expect to get a Reference Error for `foo`

**Actual behavior:** 

```js
{
    var foo = 'bar';
}
console.log(foo);

```

There is no Reference Error in emitted code and `bar` is printed to the console.

**Comment by vladima** [comment]
compiler will also signal about the mistake by producing error `a.ts(5,13): error TS2304: Cannot find name 'foo'.`. However by default we don't block the emit in presence of semantic errors (i.e user might be doing `.js` -> `.ts` migration). If you want to do emit only if there are no errors - you can use `--noEmitOnError` compiler option

**Comment by subhaze** [comment]
I don't see this as a semantic error, since all is semantically correct. I just noticed that it was "seeing" errors in code via .ts, but emitting valid JS code.

FWIW, I'm not seeing this error emitted when --allowJs is added and compiling es6 -> es3

`tsc --allowJs --outFile main-dist.js --module system main.js`

shows no errors

**Comment by vladima** [comment]
ok, this makes more sense - for `.js` files we don't report the entire set of detected semantic errors. #6802 already tracks this.
